src/azure-cli/azure/cli/command_modules/storage/operations/account.py

```diff
@@ -476,6 +499,49 @@ def update_storage_account(cmd, instance, sku=None, tags=None, custom_domain=Non
+        AccountImmutabilityPolicyProperties = cmd.get_models('AccountImmutabilityPolicyProperties')
+        immutability_policy = None
+        from azure.cli.core.azclierror import InvalidArgumentValueError
+
+        if not instance.immutable_storage_with_versioning.enabled:
+            raise InvalidArgumentValueError("Incorrect usage: The account level immutability has been disabled. "
+                                            "The property is immutable and can only be set to true at the account "
+                                            "creation time.")
+
+        existing_policy = instance.immutable_storage_with_versioning.immutability_policy
+
+        if not existing_policy:
+            if not all([immutability_period_since_creation_in_days, immutability_policy_state, allow_protected_append_writes is not None]):
+                raise InvalidArgumentValueError("Incorrect usage: To create the account level immutability policy, "
+                                                "need to specify all of the following arguments:"
+                                                "--immutability-period --immutability-state "
+                                                "--allow-append")
+        else:
+            if existing_policy.state == 'Disabled':
+                if immutability_policy_state and immutability_policy_state == 'Locked':
+                    raise InvalidArgumentValueError(
+                        "Incorrect usage: The immutability policy state can only be changed to Locked from Unlocked.")
+            if existing_policy.state == 'Locked':
+                if immutability_policy_state and immutability_policy_state != 'Locked':
+                    raise InvalidArgumentValueError("Incorrect usage: The immutability policy state cannot be changed once Locked.")
+
+                if immutability_period_since_creation_in_days and int(immutability_period_since_creation_in_days) < int(existing_policy.immutability_period_since_creation_in_days):
+                    raise InvalidArgumentValueError("Incorrect usage: Locked state only allows the increase of the immutability retention time. ")
+
+                if allow_protected_append_writes is not None and existing_policy.allow_protected_append_writes != allow_protected_append_writes:
+                    raise InvalidArgumentValueError("Incorrect usage: allow-protected-append-write cannot be changed for Locked time-based retention policies.")
+
+        immutability_policy = AccountImmutabilityPolicyProperties(
+            immutability_period_since_creation_in_days=immutability_period_since_creation_in_days or existing_policy.immutability_period_since_creation_in_days,
+            state=immutability_policy_state or existing_policy.state,
+            allow_protected_append_writes=allow_protected_append_writes if allow_protected_append_writes is not None else existing_policy.allow_protected_append_writes
+        )
```

Comment:
For `patch` request, leaving params as `None` means not to change the original value. So you don't need to patch `existing_policy` here, just leave them `None` if users don't give any value.

Reply:
Got it.